### PR TITLE
[Serve] Migrate long running serve failure release test to v2 stack

### LINKED ISSUE #32717: [Release][Serve] Migrate `long_running_serve_failure` test to v2 stack and Anyscale Jobs
### What happened + What you expected to happen

Migrate `long_running_serve_failure` test to v2 stack and Anyscale Jobs. If migration is impossible, consider rewriting the test or removing it. If there are features missing in v2 stack or/and Anyscale Jobs that make the migration impossible, please outline them in this issue.

### Versions / Dependencies

master

### Reproduction script

-

### Issue Severity

None

### LINKED ISSUE #33868: [serve] release test failure: long_running_serve_failure
### What happened + What you expected to happen

[INFO 2023-03-29 08:40:58,971] sdk_runner.py: 140  ... command still running ...(6990 seconds) ...
--
  | [INFO 2023-03-29 08:41:28,714] sdk_runner.py: 140  ... command still running ...(7020 seconds) ...
  | [INFO 2023-03-29 08:41:58,355] sdk_runner.py: 140  ... command still running ...(7050 seconds) ...
  | [INFO 2023-03-29 08:42:28,842] sdk_runner.py: 140  ... command still running ...(7080 seconds) ...
  | [INFO 2023-03-29 08:42:58,465] sdk_runner.py: 140  ... command still running ...(7110 seconds) ...
  | [INFO 2023-03-29 08:43:28,553] sdk_runner.py: 140  ... command still running ...(7140 seconds) ...
  | [INFO 2023-03-29 08:43:58,641] sdk_runner.py: 140  ... command still running ...(7170 seconds) ...
  | [INFO 2023-03-29 08:44:28,525] sdk_runner.py: 140  ... command still running ...(7200 seconds) ...
  | [ERROR 2023-03-29 08:44:58,216] glue.py: 420  Command returned non-success status: 137
  | Traceback (most recent call last):
  | File "/tmp/release-o8AoP3QZw9/release/ray_release/glue.py", line 364, in run_release_test
  | raise_on_timeout=not is_long_running,
  | File "/tmp/release-o8AoP3QZw9/release/ray_release/command_runner/sdk_runner.py", line 160, in run_command
  | raise CommandError(f"Command returned non-success status: {status_code}")
  | ray_release.exception.CommandError: Command returned non-success status: 137
  |  
  | During handling of the above exception, another exception occurred:
  |  
  | Traceback (most recent call last):
  | File "/tmp/release-o8AoP3QZw9/release/ray_release/glue.py", line 374, in run_release_test
  | raise TestCommandError(e)
  | ray_release.exception.TestCommandError: Command returned non-success status: 137
  | 🌍 Terminating cluster | 0s
  | 🚨 Handling errors | 0s
  | 📝 Reporting results | 1s
  | [INFO 2023-03-29 08:44:59,905] log.py: 11  Test long_running_serve_failure finished after 0.00 seconds. Last logs:
  |  
  | (ServeController pid=458318) INFO 2023-03-29 08:44:47,294 controller 458318 deployment_state.py:1165 - Deploying new version of deployment zbmTEWpKoY.
  | (ServeController pid=458318) INFO 2023-03-29 08:44:47,365 controller 458318 deployment_state.py:1384 - Adding 1 replica to deployment zbmTEWpKoY.
  | (ServeController pid=458318) INFO 2023-03-29 08:44:50,991 controller 458318 deployment_state.py:1148 - Deleting deployment zbmTEWpKoY.
  | (ServeController pid=458318) INFO 2023-03-29 08:44:51,067 controller 458318 deployment_state.py:1410 - Removing 1 replica from deployment 'zbmTEWpKoY'.
  | (ServeController pid=458318) INFO 2023-03-29 08:44:52,063 controller 458318 deployment_state.py:1165 - Deploying new version of deployment FdzAoWHDcT.
  | (ServeController pid=458318) INFO 2023-03-29 08:44:52,121 controller 458318 deployment_state.py:1384 - Adding 1 replica to deployment FdzAoWHDcT.
  | (ServeController pid=458318) WARNING 2023-03-29 08:44:52,683 controller 458318 deployment_state.py:651 - Actor for replica jofQxhOQoa#KfOMHa crashed, marking it unhealthy immediately.
  | (ServeController pid=458318) WARNING 2023-03-29 08:44:52,683 controller 458318 deployment_state.py:1602 - Replica jofQxhOQoa#KfOMHa of deployment jofQxhOQoa failed health check, stopping it.
  | (ServeController pid=458318) WARNING 2023-03-29 08:44:52,882 controller 458318 deployment_state.py:651 - Actor for replica rmLhyfEocl#OAYiHZ crashed, marking it unhealthy immediately.
  | (ServeController pid=458318) WARNING 2023-03-29 08:44:52,882 controller 458318 deployment_state.py:1602 - Replica rmLhyfEocl#OAYiHZ of deployment rmLhyfEocl failed health check, stopping it.
  |  

### Versions / Dependencies

master

### Reproduction script

https://buildkite.com/ray-project/release-tests-branch/builds/1494#01872d8c-3be5-45a2-9c5e-ef4b5585aec8

### Issue Severity

None

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Migrate test to v2 stack

Release test succeed: https://buildkite.com/ray-project/release-tests-pr/builds/33594#018744aa-5036-4bcd-b991-775755ab6128

## Related issue number

Closes #33868
Closes #32717

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/long_running_tests/workloads/serve_failure.py [hunk 1]
```diff
@@ -30,6 +30,11 @@
 
 RAY_UNIT_TEST = "RAY_UNIT_TEST" in os.environ
 
+# RAY_OVERRIDE_RESOURCES will prevent "num_cpus" working
+# when adding customized resource node to Cluster.
+if "RAY_OVERRIDE_RESOURCES" in os.environ:
+    del os.environ["RAY_OVERRIDE_RESOURCES"]
+
```
Comment (edoakes) [inline release/long_running_tests/workloads/serve_failure.py:37]: hmm this seems like a hack -- should we remove where it's being set from instead?

**Review by edoakes — APPROVED** [review]

**Comment by sihanwang41** [comment]
^^ @zhe-thoughts requesting for approval.

File: release/long_running_tests/workloads/serve_failure.py [hunk 2]
```diff
@@ -30,6 +30,11 @@
 
 RAY_UNIT_TEST = "RAY_UNIT_TEST" in os.environ
 
+# RAY_OVERRIDE_RESOURCES will prevent "num_cpus" working
```
Comment (shrekris-anyscale) [inline release/long_running_tests/workloads/serve_failure.py:33]:
```suggestion
# RAY_OVERRIDE_RESOURCES will prevent "num_cpus" from working
```

File: release/long_running_tests/workloads/serve_failure.py [hunk 3]
```diff
@@ -30,6 +30,11 @@
...
+# when adding customized resource node to Cluster.
```
Comment (shrekris-anyscale) [inline release/long_running_tests/workloads/serve_failure.py:34]:
```suggestion
# when adding a node with custom resources to Cluster.
```

File: release/long_running_tests/workloads/serve_failure.py [hunk 4]
```diff
@@ -30,6 +30,11 @@
...
+if "RAY_OVERRIDE_RESOURCES" in os.environ:
+    del os.environ["RAY_OVERRIDE_RESOURCES"]
```
Comment (shrekris-anyscale) [inline release/long_running_tests/workloads/serve_failure.py:36]: Could you add an assertion inside the `RandomKiller` to make sure the env var isn't set? That would let us catch any regressions in the testing setup.

**Review by shrekris-anyscale — APPROVED** [review]
Nice work! I left a few comments.

**Review by zhe-thoughts — APPROVED** [review]

**Comment by architkulkarni** [comment]
Windows `test_air_integrations` unit test failure unrelated, this PR only modifies a release test

**Pull request merged**